src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.kt

```diff
@@ -311,8 +314,42 @@ class SonarLintRulePanel(private val project: Project, private val parent: Dispo
 
             }
             SwingHelper.setHtml(securityHotspotHeaderMessage, htmlStringBuilder.toString(), JBUI.CurrentTheme.ContextHelp.FOREGROUND)
-            headerPanel.update(project, serverFindingKey, finding.status, finding.isValid, finding.file, ruleDescription.key, ruleDescription.type, finding.vulnerabilityProbability)
-        } else {
+            headerPanel.update(
+                project,
+                serverFindingKey,
+                finding.status,
+                finding.isValid,
+                finding.file,
+                ruleDescription.key,
+                ruleDescription.type,
+                finding.vulnerabilityProbability
+            )
+        } else if (finding is LiveIssue) {
+            headerPanel.update(
+                project,
+                finding.serverFindingKey,
+                finding.status,
+                finding.isValid,
+                finding.file,
+                ruleDescription.key,
+                ruleDescription.type,
+                ruleDescription.severity
+            )
+        } else if(finding is LocalTaintVulnerability){
+            val file = finding.file() ?: return;
+
+            headerPanel.update(
+                project,
+                finding.key(),
+                "status",
+                finding.isValid(),
+                file,
+                ruleDescription.key,
+                ruleDescription.type,
+                ruleDescription.severity
+            )
+        }
```

Comment:
Do you really need to have these 3 branches? LiveIssue and LocalTaintVulnerability have a common `Finding` parent so maybe you can keep a single call.

Also, in terms of responsibility, I think the switch could be located in the `RuleHeaderPanel`. We could provide a finding to this class and it's its responsibility to know what to do with it.